agent/functional_tests/testdata/taskdefinitions/devices/task-definition.json

```diff
@@ -0,0 +1,19 @@
+{
+  "family": "ecsftest-devices",
+  "containerDefinitions": [{
+    "image": "127.0.0.1:51670/ubuntu:latest",
+    "name": "exit",
+    "cpu": 10,
+    "memory": 10,
+    "linuxParameters": {
+      "devices":[
+        {
+          "hostPath": "/dev/xvda",
+          "containerPath": "/dev/sda",
+          "permissions": ["read"]
+        }
+      ]
+    },
+    "command": ["sh", "-c", "if ls /dev/sda && ! fdisk /dev/sda; then exit 42; else exit 1; fi"]
```

Comment:
Can we not do `fdisk` here? Especially because you are mounting `/dev/xvda` into the container?  Or at least `fdisk` with no flags? `fdisk -l` seems reasonable. Or any of `lsblk` or `blockdev`?

Reply:
the `fdisk` call is supposed to fail here, since only `read` permission is added to the device in container. `-l` flag can also be used here.

Reply:
Right, but it'll also fail if the device wasn't mounted. You need a check to ensure that it's mounted as well. Can't you check the permission with something like `stat --format=%a /dev/sda`?

Reply:
I think commands like `stat` and `lsblk` take information from `sysfs`, hence do not show the right permissions for the mounted devices. 

I tried the following: 

<pre>
$ ls -l /dev/xvda
brw-rw---- 1 root disk 202, 0 Jul  5 23:37 /dev/xvda
</pre>

<pre>
docker run -it --device /dev/xvda:/dev/dummy ubuntu
# ls -l /dev/dummy
brw-rw---- 1 root disk 202, 0 Oct  6 18:54 /dev/dummy
# stat --format=%a /dev/dummy
660
# fdisk /dev/dummy
Welcome to fdisk (util-linux 2.27.1).
Changes will remain in memory only, until you decide to write them.
Be careful before using the write command.
Command (m for help): q
</pre>

and 

<pre>
docker run -it --device /dev/xvda:/dev/dummy:r ubuntu
# ls -l /dev/dummy
brw-rw---- 1 root disk 202, 0 Oct  6 18:58 /dev/dummy
# stat --format=%a /dev/dummy
660
# fdisk /dev/dummy
Welcome to fdisk (util-linux 2.27.1).
Changes will remain in memory only, until you decide to write them.
Be careful before using the write command.
fdisk: cannot open /dev/dummy: Operation not permitted
</pre>

As you mentioned, we can do `ls` to check if the device is mounted and `fdisk` for permissions.